Tips and Tricks Main Chapter Landing page

chuanhao01 opened the pull request:
I was thinking of making this chapter more modular, where the main chapter's README contains the links and a short description of the tips. Also included a way to credit the tip's author (May or may not be necessary?)  
Also adding a section on how to contribute a tip.  

Decided this should be a separate PR from actually adding a tips and tricks sub-chapter.  

Let me know what you think of the format so far @woojiahao.

chuanhao01 committed:
adding Inital draft of main chapter README

chuanhao01 committed:
format table in main README

chuanhao01 committed:
adding short contribution steps for adding tips

woojiahao commented:
I like that each author is given credit for their work. I don't have any issues with the current format. If we're making the pages modular, there'll be adjustments that will have to be made to the site

chuanhao01 commented:
> I like that each author is given credit for their work. I don't have any issues with the current format. If we're making the pages modular, there'll be adjustments that will have to be made to the site

Hmm I guess we could do that, although I think we can leave it as it is for now.  
Cause I was thinking the quick reference links would be more people using markdown.  
This was based on chapter 7, because you can see all the subchapters on the website already.  

Was also gonna ask this about the chapter 7 sub-chapters as well

![image](https://user-images.githubusercontent.com/34702921/94361322-e8500080-00e5-11eb-98b7-531fdc43463e.png)

Honestly, I don't think its too big of a deal if the website is left as is.  
Maybe later on when there is more time :D

woojiahao commented:
Yup chapter 7 works already, but from my impression, I thought you were going to make small markdown files per tip and link them in the main README. Which in this case, would not register as a chapter. But we can always make it a chapter. The site already handles sub-chapters

chuanhao01 committed:
adding subchapter requirement in readme

chuanhao01 committed:
adding rule and date completed section

chuanhao01 committed:
added reference part for sub-chapter requirements

chuanhao01 commented:
Just added:  
- Adding tips as Sub-chapter requirements
- `Date Completed` section in tips quick reference table
  - `Date Completed` section requirements are also added

woojiahao commented:
Oops, sorry, been really busy lately 😬 I'll take a look through this tonight and share my comments here! Thank you for the good work @chuanhao01 👍🏻

woojiahao commented on 09-tips-and-tricks/README.md:
```diff
@@ -0,0 +1,22 @@
+# Tips and Tricks
+
+This chapter will be about some common tips and tricks you can use for your git setup.  
```
```suggestion
This chapter will be about some common tips and tricks you can use for your Git setup.  
```

woojiahao commented on 09-tips-and-tricks/README.md:
```diff
@@ -0,0 +1,22 @@
+## Contributing a tip
+
+On top of following the guidelines in the [contributing](../CONTRIBUTING.md) document for a tip's sub-chapter, you should also:  
+- Make sure the tip is added as a sub-chapter folder (You can use other tips or chapter's 7 sub-chapters as a reference)
```
```suggestion
- Make sure the tip is added as a sub-chapter folder (You can use other tips or [chapter's 7](../07-advanced-git) sub-chapters as a reference)
```

woojiahao commented on 09-tips-and-tricks/README.md:
```diff
@@ -0,0 +1,22 @@
+
+On top of following the guidelines in the [contributing](../CONTRIBUTING.md) document for a tip's sub-chapter, you should also:  
+- Make sure the tip is added as a sub-chapter folder (You can use other tips or chapter's 7 sub-chapters as a reference)
+- Make sure you add a short description, the date when the guide was finished(Inital finished date) and a link to the quick reference table above.
```
```suggestion
- Make sure you add a short description, the date when the guide was finished (initial finished date) and a link to the quick reference table above.
```

woojiahao reviewed (changes requested):
Looks good but there are some minor changes that can be made! Good work! 😄

chuanhao01 committed:
Apply suggestions from code review

Minor changes for TIps and Tricks Main chapter's folder

Co-authored-by: Woo Jia Hao <woojiahao1234@gmail.com>

chuanhao01 commented:
Hey just got around to looking at the changes from a week ago.  
Alot of things just piled up for next week, so I think commits are going to slow down for a bit. 😆 (There are like some comps and assignments)
Hope you don't mind.

chuanhao01 committed:
Some minor changes for clarity

woojiahao commented:
No worries at all! Let's take this slow and just work on this whenever is best 👍
